spec/customer_spec.rb

```diff
@@ -0,0 +1,33 @@
+require 'customer'
+require 'dishes'
+
+describe Customer do
+  it { is_expected.to respond_to :check }
+
+  before do
+    @dish = double(:dishes)
+    allow(@dish).to receive(:dishes).and_return(["Chicken = £5",
+               "Noodles = £3",
+               "Steak = £6"])
+
+  end
+
+  it 'can select a dish from the menu' do
+    subject.select(@dish.dishes[1])
+    expect(subject.customer_dishes).to include("Noodles = £3")
+  end
+
+  it 'will check the total of dishes' do
+    subject.select(@dish.dishes[1])
+    subject.select(@dish.dishes[2])
+    expect(subject.amount_of_dishes).to eq(2)
```

Comment:
I think the story is ambiguous when it says:

> I would like to check that the total I have been given matches the sum of the various dishes in my order

Please could you try to calculate the total price for the order. For example:

1 x Chicken: £5
2 x Noodles: £6
**Total price: £11**

You will need to change the way you represent the list of dishes so that you can access the dish name and the price separately.

Reply:
I have done the calculations of the dish items in the recent push and it now calculates the total price not the total amount of items. I must have misread the task but its sorted now 👍